torch/utils/flop_counter.py

```diff
@@ -119,6 +119,29 @@ def conv_backward_flop(input_shapes: List[Any], output_shapes: List[Any]) -> int
 
     return flop_count
 
+def sdpa_flop_count(query_shape, key_shape):
+    """
+    Count flops for self-attention.
+    """
+    b, h, s, d = query_shape
+    b2, h2, s2, d2 = key_shape
+    assert b == b2 and h == h2 and d == d2
+    total_flops = 0
+    # [b, h, s, d] @ [b, h, d, s2] -> [b, h, s, s2]
+    total_flops += bmm_flop([(b * h, s, d), (b * h, d, s2)])
+    # [b, h, s, s2] @ [b, h, s2, d] -> [b, h, s2, d]
+    total_flops += bmm_flop([(b * h, s, s2), (b * h, s2, d)])
```

Comment:
Not sure how precise flop_counting is but in essence, we only really care about the bmms right?

Reply:
more or less yeah.